pyppeteer/page.py

```diff
@@ -201,21 +184,89 @@ def _onTargetDetached(event: Dict) -> None:
         client.on('Runtime.exceptionThrown',
                   lambda exception: self._handleException(
                       exception.get('exceptionDetails')))
-        client.on('Security.certificateError',
-                  lambda event: self._onCertificateError(event))
         client.on('Inspector.targetCrashed',
                   lambda event: self._onTargetCrashed())
         client.on('Performance.metrics',
                   lambda event: self._emitMetrics(event))
         client.on('Log.entryAdded',
                   lambda event: self._onLogEntryAdded(event))
+        client.on('Page.fileChooserOpened',
+                  lambda event: self._onFileChooser(event))
 
         def closed(fut: asyncio.futures.Future) -> None:
-            self.emit(Page.Events.Close)
+            self.emit(Events.Page.Close)
             self._closed = True
 
         self._target._isClosedPromise.add_done_callback(closed)
 
+    async def _initialize(self):
+        await asyncio.gather(
+            self._frameManager.initiliaze(),
+            self._client.send('Target.setAutoAttach', {
+                'autoAttach': True,
+                'waitForDebuggerOnStart': False,
+                'flatten': True,
+            }
+                              ),
+            self._client.send('Performance.enable'),
+            self._client.send('Log.enable'),
+        )
+
+    async def _onFileChooser(self, event: dict):
+        if not self._fileChooserInterceptors.size:
+            return
+        frame = self._frameManager.frame(event['frameId'])
+        context = await frame.executionContext()
+        element = await context._adoptBackednNodeId(event['backednNodeId'])
+        interceptors = copy(self._fileChooserInterceptors)
+        self._fileChooserInterceptors.clear()
+        fileChooser = FileChooser(self._client, element, event)
+        for interceptor in interceptors:
+            interceptor.call(None, fileChooser)
+
+    async def waitForFileChooser(self, timeout: float = None):
+        if not self._fileChooserInterceptors.size:
+            await self._client.send(
+                'Page.setInterceptFileChooserDialog', {
+                    'enabled': True}
+            )
+        if not timeout:
+            timeout = self._timeoutSettings.timeout()
+
+        promise = self._loop.create_future()
+        promise = asyncio.Future()
+        callback = promise.result
+        self._fileChooserInterceptors.append(callback())
+        # todo double check helpers
+        try:
+            return helper.waitWithTimeout(promise, 'waiting for file chooser')
+        except Exception as e:
+            self._fileChooserInterceptors.remove(callback())
+            raise e
+
+    async def setGeolocation(
+            self,
+            longitude: float,
+            latitude: float,
+            accuracy: Optional[float]
+    ) -> None:
+        if longitude < -180 or longitude > 180:
```

Comment:
simplify comparison + make it correct
```suggestion
        if -180 >= longitude >= 180:
```